lib/scheduler/configmap.go

```diff
@@ -0,0 +1,92 @@
+package scheduler
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// GetParsedConfig returns a struct instance corresponding to the supplied
+// config type. It will be fully initialized - with both the default values of
+// the type, as well as with whatever the user had specified in the JSON
+func GetParsedConfig(name, configType string, rawJSON []byte) (result Config, err error) {
+	switch configType {
+	case constantLoopingVUsType:
+		config := NewConstantLoopingVUsConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	case variableLoopingVUsType:
+		config := NewVariableLoopingVUsConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	case sharedIterationsType:
+		config := NewSharedIterationsConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	case perVUIterationsType:
+		config := NewPerVUIterationsConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	case constantArrivalRateType:
+		config := NewConstantArrivalRateConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	case variableArrivalRateType:
+		config := NewVariableArrivalRateConfig(name)
+		err = json.Unmarshal(rawJSON, &config)
+		result = config
+	default:
+		return nil, fmt.Errorf("unknown execution scheduler type '%s'", configType)
+	}
+	return
+}
```

Comment:
Maybe we can have different schedulers add themselves to a map like the sql drivers do and than just going through the map ?

Reply:
Great idea, will do it that way!